pymatgen/io/shengbte.py

```diff
@@ -0,0 +1,275 @@
+# pymatgen>pymatgen>io
+
+# coding: utf-8
+# Copyright (c) Pymatgen Development Team.
+# Distributed under the terms of the MIT License
+
+
+import f90nml
+import numpy as np
+import re
+
+"""
+This module defines IO ShengBTE for reading, updating, and writing the
+CONTROL input file
+"""
+
+__author__ = "Rees Chang"
+__copyright__ = "Copyright 2019, The Materials Project"
+__version__ = "0.1"
+__email__ = "rc564@cornell.edu"
+__date__ = "June 27, 2019"
+
+
+class ShengBTE_CONTROL_IO:
```

Comment:
Firstly, rename this to `Control`. The fact that it is in the shengbte module means you don't need the shengbte part. Also, we only capitalise the first letter of each word in classes.

I think this could be better written as a class which:
1. Takes several dictionaries in the constructor. One for each section of the file: e.g., flags, parameters, crystal, etc. The constructor then sets the defaults if they are required.
2. Has a static `from_file` method. Which just takes a file path as input.
3. Has a static `from_dict` method, which takes a fully formatted nested dictionary as input.
3. Has a (not static) `to_file` method, which writes the file and takes a file path as input and writes the settings by accessing them through `self`.

Reply:
For required arguments like atom types, atomic positions, etc. that depend on the material, should I still set defaults or just raise an error if they are missing arguments?